### ISSUE TITLE
[BUG] - `NB_UMASK` appears not unused

### ISSUE THREAD

**Issue by krassowski** [nebari-dev/nebari#2589]
### Describe the bug

`NB_UMASK` is set in:

https://github.com/nebari-dev/nebari/blob/200099631c2f25e9d0eaa0a23b62df4cf6597651/src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/files/jupyterhub/03-profiles.py#L330-L331

but files created have `644` permission:

![image](https://github.com/user-attachments/assets/f3a6c340-dfc9-4599-ab4e-f1519783e11b)

This is implemented in `ipython_config.py`:

https://github.com/nebari-dev/nebari/blob/200099631c2f25e9d0eaa0a23b62df4cf6597651/src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/files/ipython/ipython_config.py#L1-L11

but normally this would be placed in `jupyter_server_config.py` instead. Because `ipython_config.py` is loaded later it may have no effect. This config file is copied into the filesystem by: https://github.com/nebari-dev/nebari/blob/200099631c2f25e9d0eaa0a23b62df4cf6597651/src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/configmaps.tf#L64-L74

This is related to another issue in the same file:
- https://github.com/nebari-dev/nebari/issues/2306

### Expected behavior

Dead code such as `NB_UMASK` is removed if not needed.

### OS and architecture in which you are running Nebari

2024.6.1

### How to Reproduce the problem?

Create a file using jupyter lab UI

### Compute environment

None

**Comment by Adam-D-Lewis** [comment]
Some additional docs:
- default NB_UMASK is 022 - https://jupyter-docker-stacks.readthedocs.io/en/latest/using/common.html#permission-specific-configurations

**Comment by krassowski** [comment]
In jupyter-docker-stacks, yes but I thought we are not using it and shipping custom docker images, right? That said, I am sure that `NB_UMASK` here and in docker stacks has the same "ancestry" from old JupyterHub snippets.

**Comment by Adam-D-Lewis** [comment]
> In jupyter-docker-stacks, yes but I thought we are not using it and shipping custom docker images, right? That said, I am sure that NB_UMASK here and in docker stacks has the same "ancestry" from old JupyterHub snippets.

Ah yes, you're right.  Thanks Mike, I missed that this was jupyter-docker-stacks documentation, and I was thinking I was looking at jupyterhub docs.

**Comment by krassowski** [comment]
What is the expected behaviour here? Should the custom mask be applied or removed?

**Comment by Adam-D-Lewis** [comment]
I agree that Vini's comment probably explains why that code is in there.  I agree that code shouldn't be needed at the moment

**Comment by viniciusdc** [comment]
Open discussion on UID/GID https://github.com/nebari-dev/nebari/issues/1473 and unfortunately, the original point where the UMASK was created is on a private repo; I will try to share in our next backlog meeting.
